packages/cozy-client/src/helpers.js

```diff
@@ -75,3 +75,108 @@ export const generateWebLink = ({
 
   return url.toString()
 }
+
+const uri = (
+  {
+    protocol,
+    hostname,
+    port
+  } /*: { protocol: string, hostname: string, port: string } */
+) /*: string */ => {
+  return (
+    (protocol !== '' ? `${protocol}//` : 'https://') +
+    hostname +
+    (port !== '' ? `:${port}` : '')
+  )
+}
+
+export class InvalidProtocolError extends Error {
+  constructor(url) {
+    super(`Invalid URL protocol ${url.protocol}`)
+
+    this.url = url
+  }
+}
+
+export class InvalidCozyUrlError extends Error {
+  constructor(url) {
+    super(`URL ${url.toString()} does not seem to be a valid Cozy URL`)
+
+    this.url = url
+  }
+}
+
+const wellKnownUrl = (
+  url /*: { protocol: string, hostname: string, port: string } */
+) => uri(url) + '/.well-known/change-password'
+
+const isValidOrigin = async (
+  url /*: { protocol: string, hostname: string, port: string } */
+) /*: Promise<boolean> */ => {
+  const { status } = await fetch(wellKnownUrl(url))
+
+  if (status === 200) {
+    return true
+  } else if (status === 404) {
+    throw new InvalidCozyUrlError(url)
+  } else {
+    return false
```

Comment:
refactor:
```
  if (status === 404) {
    throw new InvalidCozyUrlError(url)
  } 
  return status === 200
```

Reply:
Ah yes. I had a test on status 401 for a little while and forgot to bring back the simple comparison.